### ISSUE TITLE
Delete a domain after it has transitioned through clientHold

### ISSUE THREAD

**Issue by kitsushadow** [cisagov/getgov#808]
### Story

As an analyst
I need to delete a domain that has gone through the clientHold transition state.
so that it's removed from the .gov zone.

### Acceptance Criteria

- [ ] add a delete button (EDIT: update the existing delete domain button as needed)
- [ ] deleting can only be done if the domain is in the state HOLD and should be done using the deleted() function (currently already added to the button

_Note:_
_- the Domain application should still show_
_- the this should not delete the domain object, only the status_

### Additional Context

The delete request from a registrant will be sent via email to CISA analysts and is not in scope of this feature. The analyst will be responsible for clicking client hold first before clicking the delete button. Do not need to include a warning pop-up in this ticket, that can be added later if desired at all. The same goes for indicating to them that client hold is needed first. 

note: this can be broken up if epp centric logic is done separately. When implementing the button portion, the developer  can copy logic existing for the place hold button on the django-admin domain object.

Refinement Question: 
Suggested approaches for what happens to the application afterwards:
1. the application will still show as "approved" but that the domain management section is not viewable and the manage icon is removed. A column will be included that contains the status of the domain object
Needs a new ticket @kitsushadow 

### Issue Links

Relates to #803 
blocked by: #901
